poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -716,6 +686,45 @@ public void saveSensorAssetsToFlowgate(Map<String, Asset> exsitingSensorAssets,
       return pduAssets;
    }
 
+   public String getPositionInfo(Asset asset) {
+      ObjectMapper mapper = new ObjectMapper();
+      StringBuilder positionInfo = new StringBuilder();
+      Map<String,String> sensorAssetJustfication = asset.getJustificationfields();
+      int rackUnitNumber = asset.getCabinetUnitPosition();
+      String rackUnitInfo = null;
+      String positionFromAsset = null;
+
+      if(rackUnitNumber != 0) {
+         rackUnitInfo = FlowgateConstant.RACK_UNIT_PREFIX  + rackUnitNumber;
+         positionInfo.append(rackUnitInfo);
+         if(sensorAssetJustfication == null || sensorAssetJustfication.isEmpty()) {
+            return positionInfo.toString();
+         }
+         String sensorInfo = sensorAssetJustfication.get(FlowgateConstant.SENSOR);
+         try {
+            Map<String,String> sensorInfoMap = mapper.readValue(sensorInfo, new TypeReference<Map<String,String>>() {});
+            positionFromAsset = sensorInfoMap.get(FlowgateConstant.POSITION);
```

Comment:
what if the positionFromAsset not exsit?

Reply:
done